### PR TITLE
Add MatchedFilterSkyMaxControl class

### PR DESCRIPTION
**Pull request by spxiwh**
More precessing code ... some actual code now though!

This is the MatchedFilterSkyMaxControl class (and associated functions) that does the majority of the precessing matched filtering. It's modelled off of the original class, but in many places two filters are needed, and there is some precession-specific stuff.

### CODE REVIEW

File: pycbc/filter/matchedfilter.py [hunk 1]
```diff
@@ -414,6 +414,321 @@ def heirarchical_matched_filter_and_cluster(self, segnum, template_norm, window)
...
+                                                thresh=self.snr_threshold,
+                                                out=self.snr_mem,
+                                                hpnorm=Iplus_norm,
+                                                hcnorm=Icross_norm,
+                                                analyse_slice=stilde.analyze)
+        # FIXME: This should live further down
+        # Convert output to pycbc TimeSeries
+        delta_t = 1.0 / (self.tlen * stilde.delta_f)
+
+        snr=TimeSeries(snr, epoch=stilde._epoch, delta_t=delta_t, copy=False)
```
Comment (ahnitz) [inline pycbc/filter/matchedfilter.py]: Minor style note. Add spaces around the equals.

Comment (spxiwh) [inline pycbc/filter/matchedfilter.py]: Why? We agreed to obey PEP 8?:

https://www.python.org/dev/peps/pep-0008/

```
Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.

Yes:

def complex(real, imag=0.0):
    return magic(r=real, i=imag)
No:

def complex(real, imag = 0.0):
    return magic(r = real, i = imag)
```

Comment (spxiwh) [inline pycbc/filter/matchedfilter.py]: Or do you mean just make:

```
snr = TimeSeries(...
```

I'll change that

Comment (ahnitz) [inline pycbc/filter/matchedfilter.py]: Yeah, sorry, I mean the Timeseries one.

Comment (spxiwh) [inline pycbc/filter/matchedfilter.py]: Fixed now.

**Pull request merged**